app/views/layouts/application.html.erb

```diff
@@ -120,6 +121,7 @@
       }
     </script>
   <% end %>
+  <div id="i18n-translations" data-translations="<%= { I18n.locale.to_sym => { core: I18n::JS.translations[I18n.locale.to_sym][:core] } }.to_json %>"></div>
```

Comment:
There is a lot of magic and inefficiency out of the box with the absolute default way the `I18n.js` does the translation files. Putting it in the view and letting JS talk to it from there seems slightly cleaner for now without affecting any business logic on the JS side.